llpc/context/llpcCompiler.cpp

```diff
@@ -2155,13 +2162,16 @@ void Compiler::LinkRelocatableShaderElf(
                 return;
             }
         }
+        WriteElfFile("vs.elf", pShaderElfs[ShaderStageVertex]);
+        WriteElfFile("fs.elf", pShaderElfs[ShaderStageFragment]);
```

Comment:
This should be conditional and moved somewhere to be more generic.  Maybe a loop at the start of this function, or just before we call it, that will what will write our the shader for every stage that is not empty.  We can get suggestions on how to name thing, so we do not overwrite the files if we compile multiple pipeline.

Reply:
Removed all these debugging code. We can add them back easily later if we need it.